core/platform/search/gae_search_services.py

```diff
@@ -190,15 +117,116 @@ def _validate_list(key, value):
                 ' type %s.' % (ind, key, type(element)))
 
 
-def delete_documents_from_index(
-        doc_ids, index, retries=DEFAULT_NUM_RETRIES):
+def _string_to_sort_expressions(input_string):
+    """Returns the sorted expression of the input string.
+
+    Args:
+        input_string: str. The input string to be sorted.
+
+    Returns:
+        list(SortExpression). A list of sorted expressions.
+
+    Raises:
+        ValueError. Fields in the sort expression do not start with '+' or '-'
+            to indicate sort direction.
+    """
+    sort_expressions = []
+    s_tokens = input_string.split()
+    for expression in s_tokens:
+        if expression.startswith('+'):
+            direction = gae_search.SortExpression.ASCENDING
+        elif expression.startswith('-'):
+            direction = gae_search.SortExpression.DESCENDING
+        else:
+            raise ValueError(
+                'Fields in the sort expression must start with "+"'
+                ' or "-" to indicate sort direction.'
+                ' The field %s has no such indicator'
+                ' in expression "%s".' % (expression, input_string))
+        sort_expressions.append(
+            gae_search.SortExpression(expression[1:], direction))
+    return sort_expressions
+
+
+def _search_document_to_dict(doc):
+    """Converts and returns the search document into a dict format.
+
+    Args:
+        doc: Document. The document to be converted into dict format.
+
+    Returns:
+        dict(str, str). The document in dict format.
+    """
+    d = {'id': doc.doc_id, 'language_code': doc.language, 'rank': doc.rank}
+
+    for field in doc.fields:
+        d[field.name] = field.value
+
+    return d
+
+
+class SearchFailureError(Exception):
+    """This error is raised when a search operation fails.
+       The original_exception will point to what went wrong inside the gae sdk.
+       Other platform implementations should have a similar way of revealing
+       platform specific errors.
+    """
+
+    def __init__(self, original_exception=None):
+        super(SearchFailureError, self).__init__(
+            '%s: %s' % (type(original_exception), original_exception.message))
+        self.original_exception = original_exception
+
+
+def add_documents_to_index(documents, index):
```

Comment:
Shouldn't the API here match exactly the ES one? E.g. if the argname is index_name there, it should also be index_name here.

Could you please go through all the functions and ensure all the signatures (Args, Returns) match exactly?